gcloud-java-storage/src/main/java/com/google/gcloud/storage/testing/FakeStorageRpc.java

```diff
@@ -91,8 +89,54 @@ public StorageObject create(StorageObject object, InputStream content, Map<Optio
   @Override
   public Tuple<String, Iterable<StorageObject>> list(String bucket, Map<Option, ?> options)
       throws StorageException {
-    potentiallyThrow(options);
-    return null;
+    String preprefix = "";
+    for (Map.Entry<Option, ?> e : options.entrySet()) {
+      if (e.getKey() == StorageRpc.Option.PREFIX) {
+        preprefix = (String)e.getValue();
+        if (preprefix.startsWith("/")) {
+          preprefix = preprefix.substring(1);
+        }
+        continue;
+      } else {
+        throw new UnsupportedOperationException("Unknown option: "+e.getKey());
```

Comment:
Nit: add a space between the two strings being concatenated: `"Unknown option: " + e.getKey()`